SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -411,6 +412,12 @@ private int scheduleTasks(SingularitySchedulerStateCache stateCache, Singularity
 
     if (numMissingInstances > 0) {
       schedule(numMissingInstances, matchingTaskIds, request, state, deployStatistics, pendingRequest, maybePendingDeploy);
+
+      List<SingularityTaskId> remainingActiveTasks = new ArrayList<>(matchingTaskIds);
```

Comment:
Log here is good, just debating if this is the right place to put this. Ideally since bounce is all cleanup related this would live in the `SingularityCleaner`. Maybe it makes sense for each run of the cleaner to double check any in progress bounces or requests that are currently marked as bouncing and verify that they are correct?

Having it here will solve the current edge case we ran into, but I could see there still being more. Moving something like that to the cleaner might give us a better long term solution